admin-ui/plugins/auth-server/helper/utils.js

```diff
@@ -0,0 +1,68 @@
+// LDAP utility functions
+export function formatLdapServers(servers) {
+  if (Array.isArray(servers)) {
+    return servers.join(', ')
+  }
+  return servers
+}
+
+export function formatBaseDNs(baseDNs) {
+  if (Array.isArray(baseDNs)) {
+    return baseDNs.join(', ')
+  }
+  return baseDNs
+}
+
+export const lDapFormInitialState = (isEdit, item) =>
+  isEdit && item
+    ? item
+    : {
+        configId: '',
+        bindDN: '',
+        bindPassword: '',
+        servers: [''],
+        maxConnections: '',
+        baseDNs: [''],
+        primaryKey: '',
+        localPrimaryKey: '',
+        level: '',
+        defaultAuthnMethod: false,
+        useSSL: false,
+        useAnonymousBind: false,
+        enabled: false,
+        version: '',
+      }
+
+export function buildLdapPayload(values, userMessage) {
+  const serversArr = values.servers
+    ? String(values.servers)
+        .split(',')
+        .map((s) => s.trim())
+        .filter(Boolean)
+    : []
+  const baseDNsArr = values.baseDNs
+    ? String(values.baseDNs)
+        .split(',')
+        .map((s) => s.trim())
+        .filter(Boolean)
+    : []
+  return {
```

Comment:
Unconditionally coercing an input value to a string before parsing it is bad because it violates the best practice of defensive programming by failing to handle cases where the input is already in the desired array format, which leads to unnecessary operations and potential bugs.